libraries/AP_Mission/AP_Mission.cpp

```diff
@@ -2533,6 +2592,90 @@ bool AP_Mission::distance_to_landing(uint16_t index, float &tot_distance, Locati
     return ret;
 }
 
+// Approximate the distance travelled to return to the mission path. DO_JUMP commands are observed in look forward.
+// Stop searching once reaching a landing or do-land-start
+bool AP_Mission::distance_to_mission_leg(uint16_t start_index, float &rejoin_distance, uint16_t &rejoin_index, const Location& current_loc)
+{
+    Location prev_loc;
+    Mission_Command temp_cmd;
+    rejoin_distance = -1;
+    rejoin_index = -1;
+    bool ret = false;
+
+    // back up jump tracking to reset after distance calculation
+    jump_tracking_struct _jump_tracking_backup[AP_MISSION_MAX_NUM_DO_JUMP_COMMANDS];
+    for (uint8_t i=0; i<AP_MISSION_MAX_NUM_DO_JUMP_COMMANDS; i++) {
+        _jump_tracking_backup[i] = _jump_tracking[i];
+    }
+
+    // run through remainder of mission to approximate a distance to landing
+    uint16_t index = start_index;
+    for (uint8_t i=0; i<255; i++) {
```

Comment:
Is this 255 OK?  I guess it should be a UINT8_MAX at least if not a constant?

Reply:
Its a copy of the `distance_to_landing` method here:
https://github.com/ArduPilot/ardupilot/blob/c9f7a3c03d7ef58e328c7ebbe6b4be0d8b030394/libraries/AP_Mission/AP_Mission.cpp#L2495

255 is arbitrary really, its just a attempt to constrain the maximum possible search time.